freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/freeipa/stack/model/common/DetailedStackStatus.java

```diff
@@ -1,80 +1,99 @@
-    COLLECTING_HOST_METADATA(Status.UPDATE_IN_PROGRESS),
-    MOUNTING_DISKS(Status.UPDATE_IN_PROGRESS),
-    CONFIGURING_ORCHESTRATOR(Status.UPDATE_IN_PROGRESS),
-    VALIDATING_CLOUD_STORAGE(Status.UPDATE_IN_PROGRESS),
-    STARTING_FREEIPA_SERVICES(Status.UPDATE_IN_PROGRESS),
-    REGISTER_WITH_CLUSTER_PROXY(Status.UPDATE_IN_PROGRESS),
-    UPDATE_CLUSTER_PROXY_REGISTRATION(Status.UPDATE_IN_PROGRESS),
-    POSTINSTALL_FREEIPA_CONFIGURATION(Status.UPDATE_IN_PROGRESS),
+    BOOTSTRAPPING_MACHINES(Status.UPDATE_IN_PROGRESS, false),
+    COLLECTING_HOST_METADATA(Status.UPDATE_IN_PROGRESS, false),
+    MOUNTING_DISKS(Status.UPDATE_IN_PROGRESS, false),
+    CONFIGURING_ORCHESTRATOR(Status.UPDATE_IN_PROGRESS, false),
+    VALIDATING_CLOUD_STORAGE(Status.UPDATE_IN_PROGRESS, false),
+    STARTING_FREEIPA_SERVICES(Status.UPDATE_IN_PROGRESS, false),
+    REGISTER_WITH_CLUSTER_PROXY(Status.UPDATE_IN_PROGRESS, false),
+    UPDATE_CLUSTER_PROXY_REGISTRATION(Status.UPDATE_IN_PROGRESS, false),
+    POSTINSTALL_FREEIPA_CONFIGURATION(Status.UPDATE_IN_PROGRESS, false),
     // Start statuses
-    START_REQUESTED(Status.START_REQUESTED),
-    START_IN_PROGRESS(Status.START_IN_PROGRESS),
-    STARTED(Status.AVAILABLE),
-    START_FAILED(Status.START_FAILED),
+    START_REQUESTED(Status.START_REQUESTED, false),
+    START_IN_PROGRESS(Status.START_IN_PROGRESS, false),
+    STARTED(Status.AVAILABLE, true),
+    START_FAILED(Status.START_FAILED, false),
     // Stop statuses
-    STOP_REQUESTED(Status.STOP_REQUESTED),
-    STOP_IN_PROGRESS(Status.STOP_IN_PROGRESS),
-    STOPPED(Status.STOPPED),
-    STOP_FAILED(Status.STOP_FAILED),
+    STOP_REQUESTED(Status.STOP_REQUESTED, false),
+    STOP_IN_PROGRESS(Status.STOP_IN_PROGRESS, false),
+    STOPPED(Status.STOPPED, false),
+    STOP_FAILED(Status.STOP_FAILED, false),
     // Upscale statuses
-    UPSCALE_REQUESTED(Status.UPDATE_REQUESTED),
-    UPSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    UPSCALE_COMPLETED(Status.AVAILABLE),
-    UPSCALE_FAILED(Status.UPSCALE_FAILED),
+    UPSCALE_REQUESTED(Status.UPDATE_REQUESTED, true),
+    UPSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS, true),
+    UPSCALE_COMPLETED(Status.AVAILABLE, true),
+    UPSCALE_FAILED(Status.UPSCALE_FAILED, false),
     // Downscale statuses
-    DOWNSCALE_REQUESTED(Status.UPDATE_REQUESTED),
-    DOWNSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    DOWNSCALE_COMPLETED(Status.AVAILABLE),
-    DOWNSCALE_FAILED(Status.DOWNSCALE_FAILED),
+    DOWNSCALE_REQUESTED(Status.UPDATE_REQUESTED, true),
+    DOWNSCALE_IN_PROGRESS(Status.UPDATE_IN_PROGRESS, true),
+    DOWNSCALE_COMPLETED(Status.AVAILABLE, true),
+    DOWNSCALE_FAILED(Status.DOWNSCALE_FAILED, false),
     // Repair statuses
-    REPAIR_REQUESTED(Status.UPDATE_REQUESTED),
-    REPAIR_IN_PROGRESS(Status.UPDATE_IN_PROGRESS),
-    REPAIR_COMPLETED(Status.AVAILABLE),
-    REPAIR_FAILED(Status.REPAIR_FAILED),
+    REPAIR_REQUESTED(Status.UPDATE_REQUESTED, true),
```

Comment:
Would this not imply that when an instance is in a faulty state (and therefore is _not_ available) and a user then attempts to repair it, it would then become available? Is that okay?

Reply:
During a repair operation, at least 1 freeipa must be available otherwise there isn't a node the replicate from.

Reply:
Ah okay I did not know that, thanks for the info.